### PR TITLE
Deprecate `ThemeData.primaryX`

### PR DESCRIPTION
**Pull request by guidezpl**
In line with https://flutter.dev/go/material-theme-system-updates and https://github.com/flutter/flutter/issues/91772, deprecates redundant/unused `ThemeData` properties.

Old | New | Previous usages
--- | --- | --- 
`ThemeData.primaryColor` | `ColorScheme.primary` | too many to list
`ThemeData.primaryColorLight` |  removed | `CircleAvatar` default background
`ThemeData.primaryColorDart` |  removed | `CircleAvatar` default background

A [Dart fix](https://dart.dev/tools/dart-fix) is provided. 

This is not a breaking change, per our [policy](https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes).

This PR also improves the description of `ThemeData.colorScheme` and re-orders a few already-deprecated `ThemeData` members to the right spot and fixes version number copy-paste error from https://github.com/flutter/flutter/pull/110162.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/theme_data.dart
Comment (esouthren) [inline packages/flutter/lib/src/material/theme_data.dart:296]: Small nit: [primarySwatch] ?

**Review by esouthren — APPROVED** [review]
Great stuff! LGTM.

File: packages/flutter/lib/src/material/theme_data.dart
Comment (Piinks) [inline packages/flutter/lib/src/material/theme_data.dart:1742]: Why does this getter for `bottomAppBarColor` change?

File: packages/flutter/lib/src/material/theme_data.dart
Comment (Piinks) [inline packages/flutter/lib/src/material/theme_data.dart:1713]: Same here, why does this move? Also, I think this should be deprecated already... it might be an oversight from another change. In general, since this PR is for primaryX, anything else should be done in a separate change.

File: packages/flutter/lib/src/material/theme_data.dart
Comment (Piinks) [inline packages/flutter/lib/src/material/theme_data.dart:1721]: Same here, moving things around unnecessarily makes it hard to parse the git history.

File: packages/flutter/lib/src/material/theme_data.dart
Comment (Piinks) [inline packages/flutter/lib/src/material/theme_data.dart:1922]: Why are the versions changing?

File: packages/flutter/lib/src/material/theme_data.dart
Comment (Piinks) [inline packages/flutter/lib/src/material/theme_data.dart:1927]: These also appear unrelated. Why is this version changing?

File: packages/flutter/lib/src/material/theme_data.dart
Comment (Piinks) [inline packages/flutter/lib/src/material/theme_data.dart:295]: I don't think making this a dartdoc here (`///`) will surface it in the public docs since this is just the parameter. Can you add this to the docs for ThemeData.brightness?

**Review by Piinks — CHANGES_REQUESTED** [review]
It looks like this broke a few things in flutter/plugins. Those will need to be migrated first before this can land.

**Comment by guidezpl** [comment]
Please see https://github.com/flutter/plugins/pull/6417

File: packages/flutter/lib/src/material/theme_data.dart
Comment (guidezpl) [reply packages/flutter/lib/src/material/theme_data.dart:295]: Right, but it's just to get click-to-definition working in the editors

File: packages/flutter/lib/src/material/theme_data.dart
Comment (guidezpl) [reply packages/flutter/lib/src/material/theme_data.dart:1927]: This fixes a copy-paste error from a past PR

File: packages/flutter/lib/src/material/theme_data.dart
Comment (guidezpl) [reply packages/flutter/lib/src/material/theme_data.dart:1713]: `ThemeData` is now strictly organized, these were not re-ordered in past PRs, as noted in the PR description.

File: packages/flutter/lib/src/material/theme_data.dart
Comment (guidezpl) [reply packages/flutter/lib/src/material/theme_data.dart:1721]: Moving per our strict organisation of `ThemeData`

File: packages/flutter/lib/src/material/theme_data.dart
Comment (Piinks) [reply packages/flutter/lib/src/material/theme_data.dart:1713]: Right, does all of this need to happen in one PR?

File: packages/flutter/lib/src/material/theme_data.dart
Comment (guidezpl) [reply packages/flutter/lib/src/material/theme_data.dart:1742]: It's unchanged, moved from its current incorrect location

File: packages/flutter/lib/src/material/theme_data.dart
Comment (Piinks) [reply packages/flutter/lib/src/material/theme_data.dart:1927]: Can you link the PR?

File: packages/flutter/lib/src/material/theme_data.dart
Comment (guidezpl) [reply packages/flutter/lib/src/material/theme_data.dart:1713]: I don't think it needs to happen in separate PRs, we often fix things as we go. I can edit the title to reflect that

File: packages/flutter/lib/src/material/theme_data.dart
Comment (guidezpl) [reply packages/flutter/lib/src/material/theme_data.dart:1927]: https://github.com/flutter/flutter/pull/110162, linked in description

**Comment by guidezpl** [comment]
@Piinks How does this look, especially the fixes?

**Review by Piinks — COMMENTED** [review]
Fixes look good. I still think this should be split in 2 at least. The clean up is great, but lumping it in with deprecating something means all of it could end up getting reverted if the deprecation breaks something internally. It is not uncommon for deprecations to get reverted.

 Further, from the point of view of removing deprecations later, it is a lot easier if the git history is clean.

**Comment by guidezpl** [comment]
That makes sense, I'll land the fixes in a separate PR.

**Review by Piinks — APPROVED** [review]
LGTM once CI is happy. flutter/plugins probably just hasn't rolled in your change yet.

**Pull request closed without merging**
